cpp/src/arrow/util/basic_decimal.h

```diff
@@ -54,6 +61,20 @@ class ARROW_EXPORT BasicDecimal128 {
       : high_bits_(high), low_bits_(low) {}
 #endif
 
+  /// \brief Create a BasicDecimal256 from the two's complement representation.
+  /// Input array is assumed to be in native endianness.
+#if ARROW_LITTLE_ENDIAN
+  constexpr BasicDecimal128(const std::array<uint64_t, 2>& array) noexcept
+      : low_bits_(array[0]), high_bits_(static_cast<int64_t>(array[1])) {}
+#else
+  constexpr BasicDecimal128(const std::array<uint64_t, 2>& array) noexcept
+      : high_bits_(static_cast<int64_t>(array[0])), low_bits_(array[1]) {}
+#endif
+
+  // XXX
+  BasicDecimal128(LittleEndianArrayTag, const std::array<uint64_t, 2>& array) noexcept
+      : BasicDecimal128(BitUtil::LittleEndianArray::ToNative(array)) {}
```

Comment:
@bkietz What do you think about this constructor scheme? Is it ok, or is there a better option?

Reply:
I'd say it's not idiomatic arrow; I'd expect something like:
```suggestion
  static BasicDecimal128 FromLittleEndianArray(const std::array<uint64_t, 2>& array) noexcept {
    return {BitUtil::LittleEndianArray::ToNative(array)};
  }
```
... but then we'll have to repeat the declaration for non Basic decimal types.

Reply:
Well, yes, having to repeat the declaration is the problem here.